cardigan/stories/global/palette/palette.stories.tsx

```diff
@@ -2,27 +2,55 @@ import { FunctionComponent } from 'react';
 import { themeValues } from '@weco/common/views/themes/config';
 import styled from 'styled-components';
 import { font } from '@weco/common/utils/classnames';
+import Divider from '@weco/common/views/components/Divider/Divider';
+
+type ColorsObject = {
+  [name: string]: Category;
+};
+
+type Category = {
+  name: string;
+  description: string;
+  colors?: ColorObject[];
+};
+
+type ColorObject = {
+  hex: string;
+  rgb: { r: string; g: string; b: string };
```

Comment:
Shouldn't these `string`s be `number`s?

Reply:
That is super weird, I've done some tests and it really doesn't seem to care. I'll switch them, but it seems a bit useless. What @alexwlchan says might make sense... If I type them separately and tell the function what to expect as an output, then it checks, but I'm thinking it might be overkill in this case?

Reply:
I think the issue is the massive object with all the unpacking; if you break it up and add subtypes, the issue falls out.

Reply:
Yeah that's what I meant by typing separately - subtypes! Just doing that doesn't flag anything either though, I think it needs that AND adding an expected output (which I guess we should've had anyway!), only then does it start flagging errors for me.
So like:
`function hexToRgb(hex): RGB {[...]}`

Feels like a lot for this, but otherwise our typing isn't useful, so I'll make a PR.

That was an interesting one, thanks @davidpmccormick !